shared/subprocess/manager.go

```diff
@@ -22,6 +23,18 @@ func NewProcess(name string, args []string, stdoutPath string, stderrPath string
 	return &proc, nil
 }
 
+// NewProcessWithFds is a constructor for a process object. Represents a process with argument config. Returns an address to process
+func NewProcessWithFds(name string, args []string, stdin io.Reader, stdout io.Writer) (*Process, error) {
```

Comment:
Would it be possible now to remove the `Stdout` and `Stderr` fields from `Process` and modify the `NewProcess` function to perform the logic that is currently done in `start()`:

```go
	// Setup output capture.
	if p.Stdout != "" {
		out, err := os.Create(p.Stdout)
		if err != nil {
			return errors.Wrapf(err, "Unable to open stdout file")
		}
		defer out.Close()
		cmd.Stdout = out
	}

	if p.Stderr == p.Stdout {
		cmd.Stderr = cmd.Stdout
	} else if p.Stderr != "" {
		out, err := os.Create(p.Stderr)
		if err != nil {
			return errors.Wrapf(err, "Unable to open stderr file")
		}
		defer out.Close()
		cmd.Stderr = out
	}
```

which would then set the file handles for the opened/created files into `StdinReader` and `StdoutWriter`?
This would need to accept `io.ReadCloser` rather `io.Reader` and `io.WriteClose` respectively so we can still close the handles from `start()`.

Reply:
`NewProcessWithFds` accepts `io.ReadCloser` and `io.WriteClose` respectively, right now. I've modified also `NewProcess` which is using `NewProcessWithFds`.